packages/core/src/types.ts

```diff
@@ -1254,6 +1254,7 @@ export interface SendActionObject<
     event: TSentEvent;
     delay?: number;
     id: string | number;
+    internal: boolean;
```

Comment:
Is this needed or can it be derived from `to` being self?

There's a possibility for an impossible state if `to` is an external actor but `internal: true`

Reply:
> Is this needed or can it be derived from to being self?

It can't be derived right now because it's still valid to send an external event to `self`. Perhaps this should be changed and all events sent to itself should be raised but that would be a separate change.

> There's a possibility for an impossible state if to is an external actor but internal: true

Right, however, we are in full control when it comes to creating those objects and we just don't allow this invalid combination.